src/main/scala/com/gu/acquisition/services/GAService.scala

```diff
@@ -69,6 +76,14 @@ private[services] class GAService(implicit client: OkHttpClient)
       .mkString("&")
   }
 
+  private[services] def sanitiseClientId(maybeId: String) = {
+    maybeId match {
+      case clientIdPattern(id) => id // If we have a full _ga cookie string extract the client id
+      case "" => UUID.randomUUID().toString // If the provided value is blank we need to create one
```

Comment:
can `maybeId` be an `Option[String]` to avoid the empty string comparison?

Reply:
> that will imply to callers that it's ok to pass None to this which it isn't

Unless we throw an exception when `maybeId` = `""` then we're saying it is ok imo.